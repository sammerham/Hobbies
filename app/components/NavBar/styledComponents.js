import styled from 'styled-components';
import { Link } from 'react-router-dom';

// Create a navbar component that'll render an <div> tag with some styles

export const StyledNavBar = styled.div`
  background-color: ${props => props.bg};
  display: flex;
  padding: 1rem;
  width: 100%;
`;

// Create a link component that'll render an <Link> tag with some styles

export const StyledLink = styled(Link)`
  text-decoration: none;
  margin: 0.5rem;
  color: ${props => props.bg};
  &:hover {
    color: steelblue;
  }
`;
