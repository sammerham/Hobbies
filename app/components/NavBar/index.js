import React from 'react';
import { StyledNavBar, StyledLink } from './styledComponents';

/*
 * NavBar
 * App-- -> NavBar
 */

export default function NavBar() {
  return (
    <StyledNavBar bg="lightgrey">
      <StyledLink to="/" bg="black">
        Home
      </StyledLink>
      <StyledLink to="/new" bg="black">
        New Hobby
      </StyledLink>
    </StyledNavBar>
  );
}
