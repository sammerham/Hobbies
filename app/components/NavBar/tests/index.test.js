/**
 *
 * Tests for NavBar
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import NavBar from '../index';

const renderer = new ShallowRenderer();
describe('<NavBar />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<NavBar />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
