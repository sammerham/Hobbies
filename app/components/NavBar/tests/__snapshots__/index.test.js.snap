// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render and match the snapshot 1`] = `
<ForwardRef
  bg="lightgrey"
>
  <ForwardRef
    bg="black"
    to="/"
  >
    Home
  </ForwardRef>
  <ForwardRef
    bg="black"
    to="/new"
  >
    New Hobby
  </ForwardRef>
</ForwardRef>
`;
